### DISCUSSION TITLE
How add icon in default statusline?

### DISCUSSION THREAD

**Discussion by DinkyTrady** [kyazdani42/nvim-web-devicons#164] [Q&A] [answered]
I want to add icon in default statusline without using statusline plugin because if i used it will take longer for open file

**Comment by T-727** [comment, marked as answer]
This's a fairly open-ended question but here's what I can tell you:

1. statusline plugins are usually pretty optimized and shouldn't slowdown opening files
2. adding the icon is pretty much replicating what a statusline plugin does but on a smaller scale

here're some examples if you want to get started:
* Easiest, but has unnecessary updates :
```vim
vim.o.statusline = [[%<%f %{luaeval('require"nvim-web-devicons".get_icon_by_filetype(vim.bo.filetype)')}%h%m%r%=%-14.(%l,%c%V%) %P]]
```
* Less updates, but slightly more complicated:
```lua
vim.api.nvim_create_autocmd("BufEnter", {
    callback = function ()
    vim.o.statusline =
    "%<%f "..
    require"nvim-web-devicons".get_icon_by_filetype(vim.bo.filetype)..
    "%h%m%r%=%-14.(%l,%c%V%) %P"
    end
})
```

**Comment by DinkyTrady** [comment]
@T-727 and how to change color?

**Reply by T-727** [reply]
You can add a highlight group to the statusline by adding `%#group_name#`,
then update the color of it with the icon.
Here's what I mean using the second example:
```lua
vim.cmd "hi StatuslineIcon gui=reverse"
vim.api.nvim_create_autocmd("BufEnter", {
    callback = function()
        local icon, color = require "nvim-web-devicons".get_icon_color_by_filetype(vim.bo.filetype)
        vim.cmd("hi StatuslineIcon guibg=" .. color)
        vim.o.statusline =
        "%<%f %#StatuslineIcon#" ..
        icon ..
        "%##%h%m%r%=%-14.(%l,%c%V%) %P"
    end
})
```
You can type `:h statusline` for more information.
